Runtime/Plugins/WebGL/MAMInterface.jslib

```diff
@@ -1,30 +1,106 @@
+            if (!window.AppState) return;
             var returnStr = window.AppState.appId;
             var buffer = _malloc(lengthBytesUTF8(returnStr) + 1);
-            var bufferSize = lengthBytesUTF8(returnStr) + 1; 
+            var bufferSize = lengthBytesUTF8(returnStr) + 1;
             stringToUTF8(returnStr, buffer, bufferSize);
             return buffer;
         },
 
         MAMGetInitialPosition: function () {
-            if(!window.AppState) return;
+            if (!window.AppState) return;
             var returnStr = window.AppState.initialPosition;
             var buffer = _malloc(lengthBytesUTF8(returnStr) + 1);
-            var bufferSize = lengthBytesUTF8(returnStr) + 1; 
+            var bufferSize = lengthBytesUTF8(returnStr) + 1;
             stringToUTF8(returnStr, buffer, bufferSize);
             return buffer;
         },
 
         MAMGetModelLoader: function () {
-            if(!window.AppState) return;
+            if (!window.AppState) return;
             var returnStr = window.AppState.modelLoader;
             var buffer = _malloc(lengthBytesUTF8(returnStr) + 1);
-            var bufferSize = lengthBytesUTF8(returnStr) + 1; 
+            var bufferSize = lengthBytesUTF8(returnStr) + 1;
             stringToUTF8(returnStr, buffer, bufferSize);
             return buffer;
         },
+
+        /**
+         * WORK IN PROGRESS
+         * @param appUrl
+         * @param transform
+         */
+        MAMLaunchApp: function (appUrl, transform) {
+            if (!window.AppState) return;
+            console.log("** launching ** ", appUrl);
+            window.parent.dispatchEvent(
+                new CustomEvent("plutomae-xrpk-event", {
+                    detail: {
+                        type: "launch-xrpk",
+                        appId: window.AppState.appId,
+                        url: Pointer_stringify(appUrl),
+                        transform: JSON.parse(Pointer_stringify(transform)),
+                    },
+                })
+            );
+        },
+
+        MAMLaunchAppByNameId: function (name, id, transform) {
+            if (!window.AppState) return;
+
+            console.log("** launching ** ", name);
+
+            window.parent.dispatchEvent(
+                new CustomEvent("plutomae-xrpk-event", {
+                    detail: {
+                        type: "launch-xrpk-by-name-id",
+                        name: Pointer_stringify(name),
+                        uuid: Pointer_stringify(id),
+                        transform: JSON.parse(Pointer_stringify(transform)),
+                    },
+                })
+            );
+        },
+
+        MAMGetApps: function (receiverName, dataMethodName, errorMethodName) {
+            if (!window.AppState) return;
+
+            console.log("### in MAMGetApps")
+            console.log("### unityInstance ", unityInstance)
+            console.log("### window parent unityInstance ", window.parent.unityInstance)
```

Comment:
Do these logs still need to be here?

Reply:
Oh! No. They are gross. I will leave them on the curb.

Reply:
Updated!